src/utils/stackedUtils.ts

```diff
@@ -0,0 +1,179 @@
+        });
+
+        var stackOffsets = this._generateStackOffsets(
+          datasets,
+          this._stack(positiveDataMapArray, domainKeys),
+          this._stack(negativeDataMapArray, domainKeys),
+          keyAccessor,
+          valueAccessor);
+
+        return stackOffsets;
+      }
+
+      /**
+       * Calculates an extent across all datasets. The extent is a <number> interval that
+       * accounts for the fact that stacked bits have to be added together when calculating the extent
+       *
+       * @return {[number]} The extent that spans all the stacked data
+       */
+      public static computeStackExtent(
+          datasets: Dataset[],
+          keyAccessor: Accessor<any>,
+          valueAccessor: Accessor<any>,
+          stackOffsets: Utils.Map<Dataset, D3.Map<number>>,
+          filter: Accessor<boolean>) {
+
+        var maxStackExtent = Utils.Methods.max<Dataset, number>(datasets, (dataset: Dataset) => {
+          var data = dataset.data();
+          if (filter != null) {
+            data = data.filter((d, i) => filter(d, i, dataset));
+          }
+          return Utils.Methods.max<any, number>(data, (datum: any, i: number) => {
+            return +valueAccessor(datum, i, dataset) +
+              stackOffsets.get(dataset).get(String(keyAccessor(datum, i, dataset)));
+          }, 0);
+        }, 0);
+
+        var minStackExtent = Utils.Methods.min<Dataset, number>(datasets, (dataset: Dataset) => {
+          var data = dataset.data();
+          if (filter != null) {
+            data = data.filter((d, i) => filter(d, i, dataset));
+          }
+          return Utils.Methods.min<any, number>(data, (datum: any, i: number) => {
+            return +valueAccessor(datum, i, dataset) +
+              stackOffsets.get(dataset).get(String(keyAccessor(datum, i, dataset)));
+          }, 0);
+        }, 0);
+
+        return [Math.min(minStackExtent, 0), Math.max(0, maxStackExtent)];
+      }
+
+      /**
+       * Given an array of datasets and the accessor function for the key, computes the
+       * set reunion (no duplicates) of the domain of each dataset.
+       */
+      public static getDomainKeys(datasets: Dataset[], keyAccessor: Accessor<any>) {
+        var domainKeys = d3.set();
+        datasets.forEach((dataset) => {
+          dataset.data().forEach((datum, index) => {
+            domainKeys.add(keyAccessor(datum, index, dataset));
+          });
+        });
+
+        return domainKeys.values();
+      }
+
+      /**
+       * Feeds the data through d3's stack layout function which will calculate
+       * the stack offsets and use the the function declared in .out to set the offsets on the data.
+       */
+      private static _stack(dataArray: D3.Map<StackedDatum>[], domainKeys: string[]) {
+        var outFunction = (d: StackedDatum, y0: number, y: number) => {
+          d.offset = y0;
+        };
+
+        d3.layout.stack()
+          .x((d) => d.key)
+          .y((d) => +d.value)
```

Comment:
It looks more like a feature to me, and I'd like to scope it out into another issue